torch/csrc/lazy/core/shape_inference.cpp

```diff
@@ -289,6 +291,34 @@ std::vector<Shape> compute_shape_embedding_dense_backward(const at::Tensor& grad
   return {Shape(grad_output.scalar_type(), {num_weights, grad_output.size(-1)})};
 }
 
+std::vector<Shape> compute_shape_expand(const at::Tensor & self, at::IntArrayRef size, bool implicit) {
+  CHECK_GE(size.size(), self.dim());
+  int64_t num_new_dimensions = size.size() - self.dim();
+  std::vector<int64_t> padded_self(num_new_dimensions, 0);
+  padded_self.insert(padded_self.end(), self.sizes().begin(),
+                     self.sizes().end());
+  std::vector<int64_t> target_size(size.size());
+  for (const auto idx : c10::irange(size.size())) {
+    target_size[idx] = size[idx] == -1 ? padded_self[idx] : size[idx];
+  }
+  return {Shape(self.scalar_type(), target_size)};
+}
+
+std::vector<Shape> compute_shape_expand(const at::Tensor & self, c10::SymIntArrayRef size, bool implicit) {
```

Comment:
how do we determine which ops to write SymIntArrayRef versions of?  Is it (a) all ops that take size inputs, (b) only ops that we expect to consume outputs of dynamic ops like nonzero in models? or something else

Reply:
>  (b) only ops that we expect to consume outputs of dynamic ops like nonzero in models?

I believe

Reply:
> ok. relatedly- how does it manifest when one is missing? presumably it would show up as a compile error.

I think it would show up as a runtime error when a dispatcher wouldn't be able to find a dispatch that takes a version with SymInts.

> Who actually calls the 'compute_shape_expand' variant with symints?

The dispatcher. When a user passes sym ints either in python or c++, the dispatcher should dispatch to the expand overload that takes symints?